plugins/janus_audiobridge.c

```diff
@@ -6819,6 +6916,99 @@ static void *janus_audiobridge_handler(void *data) {
 	JANUS_LOG(LOG_VERB, "Leaving AudioBridge handler thread\n");
 	return NULL;
 }
+static void janus_audiobridge_rec_add_wav_header(janus_audiobridge_room *audiobridge) {
+
+	/* Do we need to record the mix? */
+	char filename[255];
+	gint64 now = janus_get_real_time();
+	audiobridge->rec_start_time = now;
+	if(audiobridge->record_file) {
+		g_snprintf(filename, 255, "%s-%"SCNi64"%s%s", audiobridge->record_file,
+			now, rec_tempext ? "." : "", rec_tempext ? rec_tempext : "");
+	} else {
+		g_snprintf(filename, 255, "janus-audioroom-%s-%"SCNi64".wav%s%s", audiobridge->room_id_str,
+			now, rec_tempext ? "." : "", rec_tempext ? rec_tempext : "");
+	}
+	audiobridge->recording = fopen(filename, "wb");
+	if(audiobridge->recording == NULL) {
+		JANUS_LOG(LOG_WARN, "Recording requested, but could NOT open file %s for writing...\n", filename);
+	} else {
+		JANUS_LOG(LOG_VERB, "Recording requested, opened file %s for writing\n", filename);
+		/* Write WAV header */
+		wav_header header = {
+			{'R', 'I', 'F', 'F'},
+			0,
+			{'W', 'A', 'V', 'E'},
+			{'f', 'm', 't', ' '},
+			16,
+			1,
+			audiobridge->spatial_audio ? 2 : 1,
+			audiobridge->sampling_rate,
+			audiobridge->sampling_rate * 2,
+			2,
+			16,
+			{'d', 'a', 't', 'a'},
+			0
+		};
+		if(fwrite(&header, 1, sizeof(header), audiobridge->recording) != sizeof(header)) {
+			JANUS_LOG(LOG_ERR, "Error writing WAV header...\n");
+		}
+		fflush(audiobridge->recording);
+		audiobridge->record_lastupdate = janus_get_monotonic_time();
+		g_atomic_int_set(&audiobridge->wav_header_added, 1);
+	}
+}
+static void janus_audiobridge_update_wav_header(janus_audiobridge_room *audiobridge) {
+
+	/* Update the length in the header */
+	fseek(audiobridge->recording, 0, SEEK_END);
+	long int size = ftell(audiobridge->recording);
+	if(size >= 8) {
+		size -= 8;
+		fseek(audiobridge->recording, 4, SEEK_SET);
+		fwrite(&size, sizeof(uint32_t), 1, audiobridge->recording);
+		size += 8;
+		fseek(audiobridge->recording, 40, SEEK_SET);
+		fwrite(&size, sizeof(uint32_t), 1, audiobridge->recording);
+		fflush(audiobridge->recording);
+	}
+	fclose(audiobridge->recording);
+	audiobridge->recording = NULL;
+	g_atomic_int_set(&audiobridge->wav_header_added, 0);
+
+	char filename[255];
+	if(audiobridge->record_file) {
+		g_snprintf(filename, 255, "%s-%"SCNi64, audiobridge->record_file,audiobridge->rec_start_time);
+	} else {
+		g_snprintf(filename, 255, "janus-audioroom-%s-%"SCNi64".wav", audiobridge->room_id_str,
+			audiobridge->rec_start_time);
+	}
+	if(rec_tempext) {
+		/* We need to rename the file, to remove the temporary extension */
+		char extfilename[255];
+		if(audiobridge->record_file) {
+			g_snprintf(extfilename, 255, "%s-%"SCNi64".%s", audiobridge->record_file, 
+				audiobridge->rec_start_time,rec_tempext);
```

Comment:
Same here, no need for the start time. You can refer to how the code was before, to see how we handled that case (`"%s%s%s"`).